pkg/kubelet/kubelet_pods.go

```diff
@@ -414,6 +414,47 @@ func (kl *Kubelet) makeEnvironmentVariables(pod *v1.Pod, container *v1.Container
 		return result, err
 	}
 
+	var (
+		configMaps = make(map[string]*v1.ConfigMap)
+		tmpEnv     = make(map[string]string)
+	)
+
+	// Env will override EnvFrom variables.
+	// Process EnvFrom first then allow Env to replace existing values.
+	for _, envFrom := range container.EnvFrom {
+		if envFrom.ConfigMapRef != nil {
+			name := envFrom.ConfigMapRef.Name
+			configMap, ok := configMaps[name]
+			if !ok {
+				if kl.kubeClient == nil {
+					return result, fmt.Errorf("Couldn't get configMap %v/%v, no kubeClient defined", pod.Namespace, name)
+				}
+				configMap, err = kl.kubeClient.Core().ConfigMaps(pod.Namespace).Get(name, metav1.GetOptions{})
+
+				if err != nil {
+					return result, err
+				}
+				configMaps[name] = configMap
+			}
+			for k, v := range configMap.Data {
+				if errMsgs := utilvalidation.IsCIdentifier(k); len(errMsgs) != 0 {
+					return result, fmt.Errorf("Invalid environment variable name, %v, from configmap %v/%v: %s", k, pod.Namespace, name, errMsgs[0])
+				}
+
+				if len(envFrom.Prefix) > 0 {
+					k = envFrom.Prefix + k
```

Comment:
> I cannot imagine a case where you merge 2 c identifiers create an invalid c identifier. But it would be more clear.

Other way around. A prefix could make a key that is not a valid C identifier (like `123`) valid (e.g. `FOO_123`).